src/HomologicalAlgebra/BoundedCocomplexes.jl

```diff
@@ -0,0 +1,392 @@
+#
+# CochainComplexes for Abelian groups.
+#
+# This should be used with any type of Abelian groups such as 
+# finite ℤ-modules, vector spaces over a field 𝕜, or finitely generated 
+# modules over a ring R. To this end we require the input to satisfy 
+# the common requirements of an Abelian category. 
+#
+#   * The cochain groups are supposed to be of type `GroupType` which 
+#     can be anything. 
+#     Note that there can be many different concrete types for one 
+#     category. For instance, for the category of modules over a ring 
+#     R, it is customary to have types for free modules, 
+#     for finitely presented modules, for submodules, etc. 
+#     We will assume that for every chain cocomplex C, all entries have 
+#     the same type. But different chain cocomplexes may have entries 
+#     of different types.
+#   * There must be a type `GroupHomType` for homomorphisms from  
+#     objects of type `GroupType1` and `GroupType2`. For the boundary 
+#     maps in one cocomplex the latter two types are assumed to 
+#     be the same.
+#   * It must be possible to compose homomorphisms of type 
+#     `GroupHomType`, i.e. there must be a function 
+#
+#       compose(::GroupHomType1, ::GroupHomType2).
+#
+#   * There must be a constructor for the zero object of the given 
+#     category. Unfortunately, we can not derive this from a type 
+#     directly, but we must provide another concrete instance of 
+#     an object in the same category, i.e.
+#
+#       zero_object(::GroupType)::GroupType
+#
+#     must be implemented for every `GroupType` that is being used.
+#   * For any concrete instances of `GroupType1` and `GroupType2` 
+#     there must be a constructor for the zero morphism:
+#
+#       zero_morphism(::GroupType1, ::GroupType2)
+#
+#   * For a homomorphism f : C → D with C of type `GroupType1` 
+#     and D of type `GroupType2` there must be the functionality 
+#
+#       kernel(f)::KernelType
+#       image(f)::ImageType
+#
+#     returning the kernel and the image as subgroups of C and D, 
+#     respectively.
+#   * For a group Z and a subgroup B ⊂ Z there must be the function
+#
+#       quo(Z, B)::QuotientType
+#
+#     returning the quotient Z/B.
+#
+#   * There must be direct sums a vector of groups:
+#
+#       direct_sum(::Vector{GroupType})
+#
+#     for all types of groups that are being used. 
+#     For a vector c = (C₁,…,Cₙ) this must return a triple 
+#
+#       (C₁ ⊕ … ⊕ Cₙ, (i₁,…,iₙ), (p₁,…, pₙ))
+#
+#     consisting of the direct sum C₁ ⊕ … ⊕ Cₙ, the injections 
+#
+#       iₖ : Cₖ ↪ C₁ ⊕ … ⊕ Cₙ
+#
+#     and the projections 
+#
+#       pₖ : C₁ ⊕ … ⊕ Cₙ → Cₖ.
+#
+#   * There must be the Hom-functor for pairs of groups 
+#
+#       Hom(::GroupType1, ::GroupType2)::GroupType3
+#
+#     taking a pair (C, D) to Hom(C, D).
+#
+########################################################################
+
+import Base: last, length, first
```

Comment:
In case this PR is supposed to be update: please note that `import` statements now should go into `src/imports.jl`